### PR TITLE
OpenMPTarget: Remove OpenMPTargetExec

### PR DESCRIPTION
**Pull request by masterleinad**
Follow-up to https://github.com/kokkos/kokkos/pull/6585#discussion_r1390462744.
When moving to execution space instances the variables in `OpenMPTargetExec` have to be non-`static`.
We also move the whole `OpenMPTargetExec` into `OpenMPTargetInstance`.

### CODE REVIEW

**Comment by masterleinad** [comment]
The OpenMPTarget build passes.

**Review by dalg24 — COMMENTED** [review]
I will leave it up to @rgayatri23 but I think that the right thing to do is refactor to get rid of `OpenMPTargetExec` and align further with what was done in the OpenACC backend

**Comment by masterleinad** [comment]
Let's wait for #7174.

**Comment by masterleinad** [comment]
> I will leave it up to @rgayatri23 but I think that the right thing to do is refactor to get rid of `OpenMPTargetExec` and align further with what was done in the OpenACC backend

I pushed a commit to remove `OpenMPTargetExec`. I'm happy to propose as a follow-up, though.

**Comment by rgayatri23** [comment]
> > I will leave it up to @rgayatri23 but I think that the right thing to do is refactor to get rid of `OpenMPTargetExec` and align further with what was done in the OpenACC backend
> 
> I pushed a commit to remove `OpenMPTargetExec`. I'm happy to propose as a follow-up, though.

Thanks. I was going to do the pushing of code into `OpenMPTarget_Instance` soon. 
I am working on making it more similar to OpenACC backend.

**Comment by masterleinad** [comment]
> Thanks. I was going to do the pushing of code into `OpenMPTarget_Instance` soon.

Let me know which commits you want me to retain here.

**Comment by rgayatri23** [comment]
> > Thanks. I was going to do the pushing of code into `OpenMPTarget_Instance` soon.
> 
> Let me know which commits you want me to retain here.

I think the current changes look good. If the CI passes, I am good.

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget.hpp [hunk 1]
```diff
@@ -31,6 +31,7 @@ static_assert(false,
 #include <cstddef>
 #include <iosfwd>
 #include <OpenMPTarget/Kokkos_OpenMPTargetSpace.hpp>
+//#include <OpenMPTarget/Kokkos_OpenMPTarget_Instance.hpp>
```
Comment (dalg24) [inline core/src/OpenMPTarget/Kokkos_OpenMPTarget.hpp:34]:
```suggestion
```

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_Instance.cpp [hunk 2]
```diff
@@ -27,11 +27,11 @@
 // constructor. undef'ed at the end
 #define KOKKOS_IMPL_OPENMPTARGET_WORKAROUND
 
+#include <Kokkos_Core.hpp>
 #include <OpenMPTarget/Kokkos_OpenMPTarget.hpp>
```
Comment (dalg24) [inline core/src/OpenMPTarget/Kokkos_OpenMPTarget_Instance.cpp:31]: This looks somewhat excessive.

File: core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp [hunk 3]
```diff
@@ -749,33 +749,28 @@ namespace Impl {
...
-  static uint32_t* m_uniquetoken_ptr;
-};
+  void* m_scratch_ptr = nullptr;
+  std::mutex m_mutex_scratch_ptr;
+  int64_t m_scratch_size      = 0;
+  uint32_t* m_uniquetoken_ptr = nullptr;
+};*/
 
 }  // namespace Impl
 }  // namespace Kokkos
```
Comment (dalg24) [inline core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel.hpp:776]: Remove instead of commenting

**Review by dalg24 — COMMENTED** [review]
Did the scope of this PR change?
I don't necessarily have an issue with the new direction, just observing that it diverges from the PR title/description.

**Comment by masterleinad** [comment]
> I don't necessarily have an issue with the new direction, just observing that it diverges from the PR title/description.

Updated.

**Review by rgayatri23 — APPROVED** [review]
OpenMPTarget build passes.

**Pull request merged**